homeassistant/components/lupusec/config_flow.py

```diff
@@ -0,0 +1,138 @@
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            username = user_input[CONF_USERNAME]
+            password = user_input[CONF_PASSWORD]
+            name = user_input.get(CONF_FRIENDLY_NAME)
+
+            try:
+                errors = await validate_user_input(
+                    self.hass, host, username, password, name
+                )
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if not errors:
+                return self.async_create_entry(
+                    title=host,
+                    data={
+                        CONF_HOST: host,
+                        CONF_USERNAME: username,
+                        CONF_PASSWORD: password,
+                        CONF_FRIENDLY_NAME: name,
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(self, user_input: dict[str, Any]) -> FlowResult:
+        """Import the yaml config."""
+        self._async_abort_entries_match(user_input)
+        host = user_input[CONF_HOST]
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+        name = user_input.get(CONF_FRIENDLY_NAME)
+        try:
+            await validate_user_input(self.hass, host, username, password, name)
+        except CannotConnect:
+            return self.async_abort(reason="cannot_connect")
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            return self.async_abort(reason="unknown")
+
+        return self.async_create_entry(title=host, data=user_input)
+
+
+async def validate_user_input(hass: HomeAssistant, host, username, password, name):
+    """Validate the provided configuration."""
+    errors = {}
+
+    if not is_valid_host(host):
+        errors[CONF_HOST] = "invalid_host"
+
+    await test_host_connection(hass, host, username, password)
+
+    return errors
+
+
+def is_valid_host(host):
+    """Check if the provided value is a valid DNS name or IP address."""
+
+    if not isinstance(host, (str, bytes)):
+        return False
+    try:
+        # Try to parse the host as an IP address
+        ipaddress.ip_address(host)
+        return True
+    except ValueError:
+        # If parsing as an IP address fails, try as a DNS name
+        try:
+            ipaddress.ip_address(socket.gethostbyname(host))
+            return True
+        except (socket.herror, ValueError, socket.gaierror):
+            return False
```

Comment:
Is there a specific reason you added this check? We connect with a lot of HTTP devices and services and we never really check if the host is valid, so imo we could remove this.

Reply:
Why try to connnect if it is clear that the adress cannot be resolved?
I can also remove the check.

Reply:
Like we never do that anywhere. I think if this would be this big of a deal we would solve this somewhere global instead of per integration

Reply:
Got it. Removed.